filters/HAGFilter.cpp

```diff
@@ -215,85 +254,53 @@ void HAGFilter::filter(PointView& view)
     // Build the 2D KD-tree.
     const KD2Index& kdi = gView->build2dIndex();
 
-    // Second pass: Find Z difference between non-ground points and the nearest
+    double maxDistance2 = std::pow(m_maxDistance, 2.0);
+    // Find Z difference between non-ground points and the nearest
     // neighbor (2D) in the ground view or between non-ground points and the
     // locally-computed surface (Delaunay triangultion of the neighborhood).
     for (PointId i = 0; i < ngView->size(); ++i)
     {
         PointRef point = ngView->point(i);
+
+        // Non-ground view point for which we're trying to calc HAG
         double x0 = point.getFieldAs<double>(Id::X);
         double y0 = point.getFieldAs<double>(Id::Y);
         double z0 = point.getFieldAs<double>(Id::Z);
+
         PointIdList ids(m_count);
         std::vector<double> sqr_dists(m_count);
         kdi.knnSearch(x0, y0, m_count, &ids, &sqr_dists);
 
-        double z1 = gView->getFieldAs<double>(Id::Z, ids[0]);
+        // Closest ground point.
+        double x = gView->getFieldAs<double>(Id::X, ids[0]);
+        double y = gView->getFieldAs<double>(Id::Y, ids[0]);
+        double z = gView->getFieldAs<double>(Id::Z, ids[0]);
+
         assert(ids.size() > 0);
```

Comment:
Originally thought this assert should be moved before the preceding three lines of code (275-277) where it is used, but is it even worth checking? We already check that the ground view is not empty at line 250. I can't think of a way that `ids` could be empty with a non-empty ground view.